### ISSUE TITLE
cl cat should have a head and tail option

### ISSUE THREAD

**Issue by percyliang** [codalab/codalab-worksheets#236]
Proposed usage:

```
cl cat -h 100 bundle
cl cat -t 100 bundle
```

**Comment by percyliang** [comment]
This should be used in the side panel and 'cl info -v' views:

```
(50 lines of head)
...
(50 lines of tail)
```

Should be a BundleClient command for this, perhaps piggybacking off head_target (or renaming this function to be more general).

**Comment by kashizui** [comment]
Also enable this truncated version of `cat` for web terminal.

**Comment by kashizui** [comment]
Taking a shot at this, but realized `-h` conflicts with the help flag. Any other ideas for interface/argument name? (Extreme possibility is entirely new commands `cl head` and `cl tail`.)

**Comment by percyliang** [comment]
How about using Python-like array indexing with --slice / -s:

cl cat -s 3:5 (lines 3 to 5)
cl cat -s :5 (first five lines)
cl cat -s :-2 (all but the last 2 lines)

**Comment by kashizui** [comment]
Hm, the current DownloadManager interface only supports head-ing and tail-ing, or slicing by bytes; we would have to add another method and introduce new query parameters to the REST API as well. But if that's okay I can go ahead and do that.

**Comment by percyliang** [comment]
Yeah, I could change the API to make it more flexible, so you can just grab
various sections of the file.  It's a bit more work since we have to
operate on lines rather than bytes...If it's too complex, we can just not
support arbitrary i:j but only do :i and -j:.

**Comment by kashizui** [comment]
Proposal then:
- Use slice notation described above for the CLI.
- Replace `head` and `tail` query parameters on `/rest/bundles/<uuid>/contents/blob/` API with more `range-start` and `range-end` parameters specifying the range of **line** indices to return.
- Update DownloadManager and worker code to support such requests. (Workers currently also take `num_head_lines` and `num_tail_lines` parameters). 

Given the amount of extra work required, I would be in support of only supporting the `:i` and `-j:` forms for now, or otherwise waiting for code to stabilize before making the bigger changes.

**Comment by kashizui** [comment]
Addressed by #785
